src/main/java/taxi/web/filters/AuthenticationFilter.java

```diff
@@ -0,0 +1,34 @@
+package taxi.web.filters;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/register")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null) {
```

Comment:
what will happen if logged in user deletes himself from the list? (your idea)

Reply:
Then `.get()` method will throw an exception

Reply:
You can process an Optional with a `isPresent` in service, returning null if no value present. I know that returning null is a bad practice, but it's better to process null value than show an exception page to the user